target_chains/starknet/contracts/src/pyth.cairo

```diff
@@ -708,4 +736,22 @@ mod pyth {
             Option::Some(i)
         }
     }
+
+    fn transfer_fee(
+        num_updates: u8,
+        caller: ContractAddress,
+        contract: ContractAddress,
+        fee_token: ContractAddress,
+        single_update_fee: u256,
+    ) -> bool {
+        let total_fee = single_update_fee * num_updates.into();
+        let fee_contract = IERC20CamelDispatcher { contract_address: fee_token };
+        if fee_contract.allowance(caller, contract) < total_fee {
+            return false;
+        }
+        if !fee_contract.transferFrom(caller, contract, total_fee) {
+            panic_with_felt252(UpdatePriceFeedsError::InsufficientFeeAllowance.into());
+        }
```

Comment:
This can fail if there is allowance but not enough balance right? I think in this situation we don't want to panic if the user has balance with the other token, right?

Reply:
Good point! The ERC20 contract actually just panics with u256_sub_overflow in this case. I've added a check of balance and a test for this case.